server/src/server.rs

```diff
@@ -158,6 +171,21 @@ impl<C: CatalogManager + 'static, Q: QueryExecutor + 'static> Builder<C, Q> {
             .build()
             .context(StartHttpService)?;
 
+        let mysql_config = mysql::MysqlConfig {
+            ip: self.config.bind_addr.clone(),
+            port: self.config.mysql_port,
+            thread_num: self.config.mysql_thread_num,
+        };
+
+        let mysql_service = mysql::Builder::new(mysql_config)
+            .runtimes(runtimes.clone())
+            .instance(instance.clone())
+            .build()
+            .context(BuildMysqlService)?;
+        if let Err(err) = mysql_service.start() {
```

Comment:
`MysqlHandler::start()` serves the TCP stream in the current thread, so this `start()` won't return until TCP is finished.

Reply:
After adding the trait constraints, maybe ceresDB can start the mysql service directly in the `build` function, I will try it later🧐